src/Raven.Server/Documents/PeriodicBackup/Azure/RavenAzureClient.cs

```diff
@@ -29,61 +29,137 @@ namespace Raven.Server.Documents.PeriodicBackup.Azure
             }
 
             RemoteFolderName = azureSettings.RemoteFolderName;
 
+            _accountName = azureSettings.AccountName;
             _containerName = azureSettings.StorageContainer;
 
             _logger = logger;
-            _serverUrlForContainer = GetUrlForContainer(azureSettings.StorageContainer.ToLower());
+            _serverUrlForContainer = GetUrlForContainer();
+            _serverUrlForAccountName = GetUrlForAccountName();
+        }
+
+        private static byte[] GetAccountKeyBytes(string accountKey)
+        {
+            try
+            {
+                return Convert.FromBase64String(accountKey);
+            }
+            catch (Exception e)
+            {
+                throw new ArgumentException($"Wrong format for {nameof(AzureSettings.AccountKey)}", e);
+            }
         }
 
-        private string GetUrlForContainer(string containerName)
+        private static void VerifySasToken(string sasToken)
+        {
+            try
+            {
+                var splitted = sasToken.Split('&');
+                if (splitted.Length == 0)
+                    throw new ArgumentException($"{nameof(AzureSettings.SasToken)} isn't in the correct format");
+
+                foreach (var keyValueString in splitted)
+                {
+                    var keyValue = keyValueString.Split('=');
+                    if (string.IsNullOrEmpty(keyValue[0]) || string.IsNullOrEmpty(keyValue[1]))
+                        throw new ArgumentException($"{nameof(AzureSettings.SasToken)} isn't in the correct format");
+                }
+            }
+            catch (Exception e)
+            {
+                throw new ArgumentException($"{nameof(AzureSettings.SasToken)} isn't in the correct format", e);
+            }
+        }
+
+        private string GetUrlForContainer()
         {
             var template = TestMode == false ? "https://{0}.blob.core.windows.net/{1}" : "http://localhost:10000/{0}/{1}";
-            return string.Format(template, _accountName, containerName);
+            return string.Format(template, _accountName, _containerName.ToLower());
         }
 
-        private string GetBaseServerUrl()
+        private string GetUrlForAccountName()
         {
             var template = TestMode == false ? "https://{0}.blob.core.windows.net" : "http://localhost:10000/{0}";
             return string.Format(template, _accountName);
         }
 
+        private string GetUrl(string baseUrl, string parameters = null)
+        {
+            var url = baseUrl;
+
+            var hasParameters = string.IsNullOrEmpty(parameters) == false;
+            if (hasParameters)
+                url += $"?{parameters}";
+
+            if (_hasSasToken)
+            {
+                if (hasParameters)
+                {
+                    url += "&";
+                }
+                else
+                {
+                    url += "?";
+                }
+
+                url += _sasToken;
```

Comment:
Do we need to encode the token?

Reply:
It is already escaped (using the Azure Portal and the Blob API)